src/plugins/search/plugin.search.js

```diff
@@ -158,15 +176,48 @@ BookReader.prototype.search = function(term = '', overrides = {}) {
         ? options.success.call(this, searchInsideResults, options)
         : this.BRSearchCallback(searchInsideResults, options);
     }
+    cleanup();
   };
 
-  this.trigger('SearchStarted', { term: this.searchTerm });
+  const beforeSend = (xhr) => {
+    this.searchXHR = xhr;
+    window.BRSearchInProgress = processSearchResults;
+  };
+
+  this.trigger('SearchStarted', { term: this.searchTerm, instance: this });
   return $.ajax({
     url: url,
-    dataType: 'jsonp'
-  }).then(processSearchResults);
+    dataType: 'jsonp',
+    beforeSend,
+    jsonpCallback: 'BRSearchInProgress'
+  }).then(processSearchResults)
 };
 
+/**
+ * cancels AJAX Call
+ * emits custom event
+ */
+BookReader.prototype._cancelSearch = function () {
+  this.searchXHR?.abort();
+  this.searchView.clearSearchFieldAndResults(false);
+  this.searchTerm = '';
+  this.searchXHR = null;
+  this.searchResults = [];
+  window.BRSearchInProgress = () => {};
```

Comment:
Does `BRSearchInProgress` get used outside of this plugin? If not, can it be a class property instead of a `window` property?

Reply:
i tried to put it in the class, but jQuery's JSONP callback naming (tho it says in [docs](https://api.jquery.com/jquery.ajax/) it can take a function, it doesn't for our version of jQuery 😢 )
![image](https://user-images.githubusercontent.com/7840857/119872418-4d0f6500-bed8-11eb-8c55-b6f915a9be7f.png)